starlite/plugins/sql_alchemy.py

```diff
@@ -6,16 +6,16 @@
 from uuid import UUID
 
 from pydantic import BaseModel, Json, conint, constr, create_model
-from pydantic_factories import ModelFactory
 from typing_extensions import Type
 
-from starlite.exceptions import (
-    ImproperlyConfiguredException,
-    MissingDependencyException,
-)
+from starlite.exceptions import ImproperlyConfiguredException
+from starlite.extras import PYDANTIC_FACTORIES, SQLALCHEMY
 from starlite.plugins.base import PluginProtocol
 
-try:
+with PYDANTIC_FACTORIES:
+    from pydantic_factories import ModelFactory
```

Comment:
Perhaps these context managers can be named so that their function is more apparent at the call site, maybe `PYDANTIC_FACTORIES_AVAILABLE` or `..._INSTALLED`..

Reply:
How about a method on Feature?

```py
with PYDANTIC_FACTORIES.ensure_installed():

```

Reply:
Sure, anything along those lines.